src/ar_http_iface_server.erl

```diff
@@ -0,0 +1,957 @@
+	case safe_decode(Addr) of
+		{error, invalid} ->
+			{400, [], <<"Invalid address.">>};
+		{ok, AddrOK} ->
+			{200, [],
+				integer_to_binary(
+					ar_tx:calculate_min_tx_cost(
+						binary_to_integer(SizeInBytes),
+						ar_node:get_current_diff(whereis(http_entrypoint_node)),
+						ar_node:get_wallet_list(whereis(http_entrypoint_node)),
+						AddrOK
+					)
+				)
+			}
+	end;
+
+%% @doc Return the current hash list held by the node.
+%% GET request to endpoint /hash_list
+handle('GET', [<<"hash_list">>], _Req) ->
+	HashList = ar_node:get_hash_list(whereis(http_entrypoint_node)),
+	{200, [],
+		ar_serialize:jsonify(
+			ar_serialize:hash_list_to_json_struct(HashList)
+		)
+	};
+
+%% @doc Return the current wallet list held by the node.
+%% GET request to endpoint /wallet_list
+handle('GET', [<<"wallet_list">>], _Req) ->
+	Node = whereis(http_entrypoint_node),
+	WalletList = ar_node:get_wallet_list(Node),
+	{200, [],
+		ar_serialize:jsonify(
+			ar_serialize:wallet_list_to_json_struct(WalletList)
+		)
+	};
+
+%% @doc Share your nodes IP with another peer.
+%% POST request to endpoint /peers with the body of the request being your
+%% nodes network information JSON encoded as specified in ar_serialize.
+% NOTE: Consider returning remaining timeout on a failed request
+handle('POST', [<<"peers">>], Req) ->
+	BlockJSON = elli_request:body(Req),
+	case ar_serialize:dejsonify(BlockJSON) of
+		{Struct} ->
+			{<<"network">>, NetworkName} = lists:keyfind(<<"network">>, 1, Struct),
+			case (NetworkName == <<?NETWORK_NAME>>) of
+				false ->
+					{400, [], <<"Wrong network.">>};
+				true ->
+					Peer = elli_request:peer(Req),
+					case ar_meta_db:get({peer, ar_util:parse_peer(Peer)}) of
+						not_found ->
+							ar_bridge:add_remote_peer(whereis(http_bridge_node), ar_util:parse_peer(Peer));
+						X -> X
+					end,
+					{200, [], []}
+			end;
+		_ -> {400, [], "Wrong network"}
+	end;
+handle('POST', [<<"peers">>, <<"port">>, RawPort], Req) ->
+	BlockJSON = elli_request:body(Req),
+	{Struct} = ar_serialize:dejsonify(BlockJSON),
+	case lists:keyfind(<<"network">>, 1, Struct) of
+		{<<"network">>, NetworkName} ->
+			case (NetworkName == <<?NETWORK_NAME>>) of
+				false ->
+					{400, [], <<"Wrong network.">>};
+				true ->
+					Peer = elli_request:peer(Req),
+					Port = binary_to_integer(RawPort),
+					ar_bridge:add_remote_peer(
+						whereis(http_bridge_node),
+						ar_util:parse_peer({Peer, Port})
+						),
+					{200, [], []}
+			end;
+		_ -> {400, [], "Wrong network"}
+	end;
+
+%% @doc Return the balance of the wallet specified via wallet_address.
+%% GET request to endpoint /wallet/{wallet_address}/balance
+handle('GET', [<<"wallet">>, Addr, <<"balance">>], _Req) ->
+	case safe_decode(Addr) of
+		{error, invalid} ->
+			{400, [], <<"Invalid address.">>};
+		{ok, AddrOK} ->
+			case ar_node:get_balance(whereis(http_entrypoint_node), AddrOK) of
+				node_unavailable ->
+					{504, [], <<"Temporary timeout.">>};
```

Comment:
`ar_node:get_balance/2` returns `node_unavailable` when there is a internal timeout. First of all, I don't see why there is a timeout at all. Elli will timeout anyway if this is too slow, so I wish `ar_node:get_balance/2` waited infinitely instead.

504 is a gateway timeout, so it doesn't sound like a good fit for this response (since it's an internal timeout). A "503 Service Unavailable" makes sense to me to though. What do you think about a 503 instead?

Reply:
AFAICS `ar_http_iface:verify_request_to_blockshadow/1` is never used.  I removed the function as well as the export.

Reply:
503 seems fine.